packages/rest/src/spec-enhancers/consolidate.spec-enhancer.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  ApplicationConfig,
+  bind,
+  BindingScope,
+  CoreBindings,
+  inject,
+} from '@loopback/core';
+import {
+  asSpecEnhancer,
+  ISpecificationExtension,
+  isSchemaObject,
+  OASEnhancer,
+  OpenApiSpec,
+  ReferenceObject,
+  SchemaObject,
+} from '@loopback/openapi-v3';
+import debugFactory from 'debug';
+import compare from 'json-schema-compare';
+import _ from 'lodash';
+
+const debug = debugFactory('loopback:openapi:spec-enhancer:consolidate');
+
+/**
+ * This enhancer consolidates schemas into `/components/schemas` and replaces
+ * instances of said schema with a $ref pointer.
+ *
+ * Please note that the title property must be set on a schema in order to be
+ * considered for consolidation.
+ *
+ * For example, with the following schema instance:
+ *
+ * ```json
+ * schema: {
+ *   title: 'loopback.example',
+ *   properties: {
+ *     test: {
+ *       type: 'string',
+ *     },
+ *   },
+ * }
+ * ```
+ *
+ * The consolidator will copy the schema body to
+ * `/components/schemas/loopback.example` and replace any instance of the schema
+ * with a reference to the component schema as follows:
+ *
+ * ```json
+ * schema: {
+ *   $ref: '#/components/schemas/loopback.example',
+ * }
+ * ```
+ *
+ * When comparing schemas to avoid naming collisions, the description field
+ * is ignored.
+ */
+@bind(asSpecEnhancer, {scope: BindingScope.SINGLETON})
+export class ConsolidationEnhancer implements OASEnhancer {
+  name = 'consolidate';
+  disabled: boolean;
+
+  constructor(
+    @inject(CoreBindings.APPLICATION_CONFIG, {optional: true})
+    readonly config?: ApplicationConfig,
+  ) {
+    this.disabled = !(this.config?.rest?.openApiSpec?.consolidate ?? true);
```

Comment:
I prefer to use `this.disabled = this.config?.rest?.openApiSpec?.consolidate === false`. It is a bit easier to read.